martin/src/utils/utilities.rs

```diff
@@ -28,3 +31,32 @@ pub fn encode_brotli(data: &[u8]) -> Result<Vec<u8>, std::io::Error> {
     encoder.write_all(data)?;
     Ok(encoder.into_inner())
 }
+
+pub fn parse_base_path(path: &String) -> MartinResult<String> {
+    if !path.starts_with('/') {
+        return Err(BasePathError(path.to_string()));
+    }
+    if let Ok(uri) = path.parse::<Uri>() {
+        return Ok(uri.path().to_string());
+    }
+    Err(BasePathError(path.to_string()))
+}
+
+#[cfg(test)]
+pub mod tests {
+    use crate::utils::parse_base_path;
+    #[test]
+    fn test_parse_base_path() {
+        let case1 = "/".to_string();
+        assert_eq!("/", parse_base_path(&case1).unwrap());
+
+        let case2 = String::new();
+        assert!(parse_base_path(&case2).is_err());
+
+        let case3 = "/foo/bar".to_string();
+        assert_eq!("/foo/bar", parse_base_path(&case3).unwrap());
+
+        let case4 = "foo/bar".to_string();
+        assert!(parse_base_path(&case4).is_err());
```

Comment:
let case5 = "/foo/bar/".to_string();
        assert!(parse_base_path(&case5).is_err());
        // Should we remove the tail slash in parse_base_path or just return an error?

Reply:
see my answer above